FirebaseAppCheck/Sources/Core/FIRAppCheck.m

```diff
@@ -234,6 +234,21 @@ - (void)getTokenForcingRefresh:(BOOL)forcingRefresh
       });
 }
 
+- (void)getLimitedUseTokenWithCompletion:(FIRAppCheckTokenHandlerInterop)handler {
```

Comment:
I may have discovered a bug in the existing `retrieveLimitedUseToken` method below. Right now it tries to see if there is an ongoing promise and waits for that if so. However, the contract for this method is that *every invocation of this method returns a fresh token*, but the existing behaviour makes it so that it's possible for multiple invocations to receive the same one-time-use token, which would be problematic. Instead, no state should be shared between invocations, and separate invocations should each have its own isolated waiting promises. I was wondering whether this can be fixed as part of this PR.

Reply:
great catch Victor– thanks for the close review here.

Reply:
Thanks, Victor! PR #11298 has now been merged into this branch.